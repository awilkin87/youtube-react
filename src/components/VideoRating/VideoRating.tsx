import { Icon, Progress } from 'semantic-ui-react';
import styles from './VideoRating.module.scss';

export interface VideoRatingProps {
  likes: number;
  dislikes: number;
}

const VideoRating: React.FunctionComponent<Readonly<VideoRatingProps>> = function ({ likes, dislikes }) {
  const totalRatings = likes + dislikes;
  const precentLikes = totalRatings ? 100 * (likes / totalRatings) : 50;

  return (
    <div className={styles.container}>
      <div className={styles.thumbsUp}>
        <Icon name='thumbs up outline' />
        <span>{likes}</span>
      </div>
      <div className={styles.thumbsDown}>
        <Icon name='thumbs down outline' />
        <span>{dislikes}</span>
      </div>
      <Progress className={styles.ratingBar} percent={precentLikes} size='tiny' />
    </div>
  );
}

export default VideoRating;