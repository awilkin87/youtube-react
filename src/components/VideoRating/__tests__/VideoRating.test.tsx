import { shallow } from 'enzyme'
import VideoRating from '../VideoRating'

test('Render VideoRating', () => {
  const wrapper = shallow(<VideoRating likes={10} dislikes={10} />);
  expect(wrapper).toMatchSnapshot();
});
