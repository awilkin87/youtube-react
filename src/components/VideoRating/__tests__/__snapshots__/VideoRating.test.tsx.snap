// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render VideoRating 1`] = `
<div
  className="container"
>
  <div
    className="thumbsUp"
  >
    <Icon
      as="i"
      name="thumbs up outline"
    />
    <span>
      10
    </span>
  </div>
  <div
    className="thumbsDown"
  >
    <Icon
      as="i"
      name="thumbs down outline"
    />
    <span>
      10
    </span>
  </div>
  <Progress
    className="ratingBar"
    percent={50}
    size="tiny"
  />
</div>
`;
